libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1485,14 +1485,29 @@ TypePointer TypeChecker::typeCheckTypeConversionAndRetrieveReturnType(
 				);
 			}
 			else
+			{
+				SecondarySourceLocation ssl;
```

Comment:
Why did you use a secondary source location? IDEs will not be able to display this properly because there is no real location attached to the secondary location. What is the downside of just appending to the error message?

Reply:
No other reason than the "Did you mean...?" in the address conversion a few lines up is also in the secondary source location. The secondary source location here is the function call argument, though, so there is a valid location (although it's almost the same as the primary location in this case)... we can just append to the error message instead, that's fine with me.